chia/rpc/wallet_rpc_api.py

```diff
@@ -3191,119 +3176,93 @@ async def dl_latest_singleton(self, request) -> Dict:
         if self.service.wallet_state_manager is None:
             raise ValueError("The wallet service is not currently initialized")
 
-        for _, wallet in self.service.wallet_state_manager.wallets.items():
-            if WalletType(wallet.type()) == WalletType.DATA_LAYER:
-                assert isinstance(wallet, DataLayerWallet)
-                only_confirmed = request.get("only_confirmed")
-                if only_confirmed is None:
-                    only_confirmed = False
-                record = await wallet.get_latest_singleton(bytes32.from_hexstr(request["launcher_id"]), only_confirmed)
-                return {"singleton": None if record is None else record.to_json_dict()}
-
-        raise ValueError("No DataLayer wallet has been initialized")
+        only_confirmed = request.get("only_confirmed")
+        if only_confirmed is None:
+            only_confirmed = False
```

Comment:
Hm, how is `only_confirmed` related here?

Reply:
A couple points.  `request` is unhinted and thus either this method is either not analyzed or `request` is `Any`.  I think the latter since there is a `Dict` return type hint.  Even if `request` were hinted, we hint as `Dict[str, Any]` so `only_confirmed` would still be `Any`.

This is the mess that will remain even if we ever get through the existing mypy loose-checking module list.